### DISCUSSION TITLE
EPUB support

### DISCUSSION THREAD

**Discussion by Lululalu** [zed-industries/zed#23873] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I would like to be able to open and edit the internal files of an epub container, similarly to what I can do with BBEdit or Oxygen XML. 

### Environment

Zed 165.4 on an Apple Silicon Mac

**Comment by bbb651** [comment]
There's no reason this should only apply to epub as they use a standard zip as the container format, this would be really useful for other zips, and other containers files generally (e.g. `tar`, `gz`, `xz`, `lz`, `7z`), as well as combinations of container formats (e.g. `tar.gz`).

I think with a system like that extension support for additional format would be useful (but absolutely not necessary, support for a few common formats is more than enough for the vast majority of people), I've written one way to generalize it in #12042 under "Additional Idea: Virutal Filesystem" but that issue a bit old, a giant wall of text and not very actionable.
In terms of prior art, vscode extensions have support for creating virtual filesystems, and there's a [ZipFS](https://marketplace.visualstudio.com/items?itemName=arcanis.vscode-zipfs) extension that (ab)uses that, but it's not a great experience because you have to open the zip file as it's own workspace (the api is designed for remote connections, I explained this in the linked issue), which probably isn't what you want. I think Zed can improve on it a lot.
